src/frontend/src/components/TestButton/TestButton.test.tsx

```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import TestButton from "./TestButton";
+import { render, screen } from "@testing-library/react";
+
+describe("TestButton", () => {
+  it("renders correctly", () => {
+    render(<TestButton variant="red" />);
+    screen.getByText("Test Button");
```

Comment:
Here you should not use this selector but `getByRole` to be sure you render a button and this button is accessible

```suggestion
    const button = screen.getByRole('button', { name: "Test Button"});
```

Then you should test it has the required variant. For this, you can install the package `@testing-library/jest-dom` that adds several asserts to jest, you can check the doc. Here you can use [toHaveStyle](https://github.com/testing-library/jest-dom#tohavestyle)

```
expect(button).toHaveStyle('color: red');
```

And to finish you should test the component with all available variants.